src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.vb

```diff
@@ -0,0 +1,1668 @@
+  IInvocationExpression ( Function A.Method() As CustomType) (OperationKind.InvocationExpression, Type: CustomType)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Method_Minus_CustomType()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As CustomType
+        Dim i As CustomType = Nothing
+        Return -Method()'BIND:"-Method()"
+    End Function
+End Class
+
+Public Class CustomType
+    Public Shared Operator -(x As CustomType) As CustomType
+        Return x
+    End Operator
+
+    Public Shared operator +(x As CustomType) As CustomType
+        return x
+    End Operator
+
+    Public Shared operator Not(x As CustomType) As CustomType
+        return x
+    End Operator
+End CLass
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.OperatorMethodMinus) (OperatorMethod: Function CustomType.op_UnaryNegation(x As CustomType) As CustomType) (OperationKind.UnaryOperatorExpression, Type: CustomType)
+  IInvocationExpression ( Function A.Method() As CustomType) (OperationKind.InvocationExpression, Type: CustomType)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Method_Not_CustomType()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As CustomType
+        Dim i As CustomType = Nothing
+        Return Not Method()'BIND:"Not Method()"
+    End Function
+End Class
+
+Public Class CustomType
+    Public Shared Operator -(x As CustomType) As CustomType
+        Return x
+    End Operator
+
+    Public Shared operator +(x As CustomType) As CustomType
+        return x
+    End Operator
+
+    Public Shared operator Not(x As CustomType) As CustomType
+        return x
+    End Operator
+End CLass
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.OperatorMethodBitwiseNegation) (OperatorMethod: Function CustomType.op_OnesComplement(x As CustomType) As CustomType) (OperationKind.UnaryOperatorExpression, Type: CustomType)
+  IInvocationExpression ( Function A.Method() As CustomType) (OperationKind.InvocationExpression, Type: CustomType)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
```

Comment:
not sure how to get OperatorMethodTrue/False

even if I implement IsTrue/IsFalse/Or/And Operators, it doesn't get returned by GetOperation